tests/steps/test_cases_functional.py

```diff
@@ -0,0 +1,195 @@
+def test_verify_that_specific_celexs_exist():
+    """Verify that specific CELEXs exist"""
+
+
+@given(parsers.cfparse('the baseURI {base_uri:String}', extra_types=dict(String=str)))
+def the_baseuri_lam_uri(browser, scenario_context, base_uri):
+    """the baseURI LAM_URI."""
+    if base_uri == "LAM_URL":
+        from tests.config import LAM_URL
+        scenario_context["base_uri"] = LAM_URL
+
+
+@when(parsers.cfparse('I navigate to the location {pageLocation:String}', extra_types=dict(String=str)))
+def i_navigate_to_the_location_page(scenario_context, browser, pageLocation):
+    browser.get(scenario_context["base_uri"] + pageLocation)
+
+
+@then(parsers.cfparse('the resulting page contains {content:String} in the element with id {field_id:String}',
+                      extra_types=dict(String=str)))
+def the_result_page_contains(browser, scenario_context, content, field_id):
+    pass
+
+
+@then(parsers.cfparse('the page has the title {title:String}',
+                      extra_types=dict(String=str)))
+def the_result_page_contains(browser, scenario_context, title):
+    page_title = browser.title
+    assert page_title == title
+
+
+@then(parsers.cfparse('the resulting page contains {content:String} in the element with XPath {xpath:String}',
+                      extra_types=dict(String=str)))
+def the_result_page_contains(browser, scenario_context, content, xpath):
+    element = browser.find_element_by_xpath(xpath)
+    assert element.text == content
+
+
+@then(parsers.cfparse('the resulting page does not contain the element with XPath {xpath:String}',
+                      extra_types=dict(String=str)))
+def the_result_page_contains(browser, scenario_context, xpath):
+    assert browser.find_element_by_xpath(xpath).text == ''
+
+
+@then(parsers.cfparse('the field with id {field_id:String} is visible', extra_types=dict(String=str)))
+def step_impl(browser, scenario_context, field_id):
+    assert browser.find_element_by_id(field_id).is_displayed() is True
+
+
+@then(parsers.cfparse('the field with id {field_id:String} is not visible', extra_types=dict(String=str)))
+def step_impl(browser, scenario_context, field_id):
+    assert browser.find_element_by_id(field_id).is_displayed() is True
+
+
+@then(parsers.cfparse('the field with XPath {field_xpath:String} is not visible', extra_types=dict(String=str)))
+def step_impl(browser, scenario_context, field_xpath):
+    sleep(1)
+    assert browser.find_element_by_xpath(field_xpath).is_displayed() is False
+
+
+@then(parsers.cfparse('the field with XPath {field_xpath:String} is visible', extra_types=dict(String=str)))
+def step_impl(browser, scenario_context, field_xpath):
+    sleep(1)
+    assert browser.find_element_by_xpath(field_xpath).is_displayed() is True
+
+
+@then(parsers.cfparse('the element with XPath {field_xpath:String} is in the viewport', extra_types=dict(String=str)))
+def step_impl(browser, scenario_context, field_xpath):
+    sleep(1)
```

Comment:
is `1` second enough for the page to render properly?

Reply:
It is about the animation to stop.